FreeRTOS/iot-reference-esp32c3
# Add C linkage for C++ support

Add C linkage for C++ support

Description
-----------
Updates the project so that linking works when using a C++.
This is achieved by conditionally adding an `extern "C"` wrapper if the project is being built using C++.

Test Steps
-----------
- Ran `idf.py build` to ensure the project still builds.
- Changed `main.c`to `main.cpp`. Then ran `idf.py build` to ensure the project builds.

Checklist:
----------



Related Issue
-----------
#65 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
